gossipd/gossip.c

```diff
@@ -49,6 +49,170 @@
 #include <wire/wire_io.h>
 #include <wire/wire_sync.h>
 
+#define SOCKS_NOAUTH		0
+#define SOCKS_ERROR 	 0xff
+#define SOCKS_CONNECT		1
+#define SOCKS_TYP_IPV4		1
+#define SOCKS_DOMAIN		3
+#define SOCKS_TYP_IPV6		4
+#define SOCKS_V5            5
+
+
+#define MAX_SIZE_OF_SOCKS5_REQ_OR_RESP 255
+#define SIZE_OF_RESPONSE 		4
+#define SIZE_OF_REQUEST 		3
+#define SIZE_OF_IPV4_RESPONSE 	6
+#define SIZE_OF_IPV6_RESPONSE 	18
+#define SOCK_REQ_METH_LEN		3
+#define SOCK_REQ_V5_LEN			5
+#define SOCK_REQ_V5_HEADER_LEN	7
+
+/* some crufts can not forward ipv6*/
+#undef BIND_FIRST_TO_IPV6
+
+struct reaching_socks {
+
+	u8 buffer[ MAX_SIZE_OF_SOCKS5_REQ_OR_RESP ];
+  	size_t hlen;
+	in_port_t  port;
+	char *host;
+	struct reaching *reach;
+};
+
+
+struct reaching_socks reach_tor;
+
+
+
+
+static struct io_plan *connect_finish(struct io_conn *,
```

Comment:
prefer to not predeclare `static` functions: instead arrange their definitions so that dependencies are defined first.  Only use predeclaration if the functions are effectively mutually-recursive.